### DISCUSSION TITLE
How to show the playlist name in the frontend ?

### DISCUSSION THREAD

**Discussion by rodrigaule** [AzuraCast/AzuraCast#5503] [Q&A]
Hi,

Has anyone here ever thought of adding the name of the current playlist on the main page of their site ? Or even a live calendar with for example the mention "currently playing : Morning show" and "playing next : Afternoon show" ?

Do you have any idea on how to proceed ?

Thank you :)

**Comment by guiltlab** [comment]
You need to use the API and javascript. Here is the code I am using on my site. Please note that I'm not an expert at all and pieced this together from copying various sources, so it might be bad and unoptimized code, but it works.

This is using jquery, if you already have it on your page you can remove the first line. Add this to your HTML:

```
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
             <script>
         var nowPlayingTimeout;
var nowPlaying;

function loadNowPlaying() {
    $.ajax({
        cache: false,
        dataType: "json",
        url: 'https://yourwebsite.com/api/nowplaying_static/station_name.json',
        success: function(np) {
            // Do something with the Now Playing data.
            nowPlaying = np;
              $('.current-playlist span').text("" + np.now_playing.playlist);

            nowPlayingTimeout = setTimeout(loadNowPlaying, 15000);
        }
    }).fail(function() {
        nowPlayingTimeout = setTimeout(loadNowPlaying, 30000);
    });
}

$(function() {
    loadNowPlaying();
});
</script>
```

to call the function, you add the class that the JS targets, in my case it's called current-playlist and it targets a span element within, so i call it like this by applying the class to a paragraph, and then inserting `<span></span>` wherever i want the playlist's name to appear:
`    <p class="current-playlist">currently listening to <b><span></span></b></p>`

which results in

> currently listening to **%current playlist's name%**

**Comment by jimmyeao** [comment]
I have a similar script that refreshes the playlist name every 60 seconds
[Grab it here](https://deviousweb.com/2024/09/06/how-to-display-the-playlist-name-on-your-azuracast-public-page/)
